engine/src/main/java/org/hibernate/search/engine/service/beanresolver/spi/ReflectionBeanResolver.java

```diff
@@ -32,16 +33,7 @@
 
 	@Override
 	public <T> T resolve(Class<?> classOrFactoryClass, Class<T> expectedClass) {
-		Object instance;
-		try {
-			instance = classOrFactoryClass.newInstance();
-		}
-		catch (InstantiationException e) {
-			throw log.noPublicNoArgConstructor( classOrFactoryClass.getName() );
```

Comment:
I guess it's on purpose, but `log.unableToAccessClass` is not used anymore. Maybe remove the method from the logger?